weldx/util/util.py

```diff
@@ -291,23 +297,13 @@ def compare_nested(
             When a or b is not a nested structure.
 
         """
-
-        def _enter_replace_sets(p, k, v):
-            if isinstance(v, set):
-                return [], ((i, j) for i, j in enumerate(sorted(list(v))))
-            return iterutils.default_enter(p, k, v)
-
-        try:
-            a = iterutils.remap(a, enter=_enter_replace_sets)
-            b = iterutils.remap(b, enter=_enter_replace_sets)
-        except TypeError:
-            raise TypeError("either a or b are not a nested data structure.")
-
         # we bind the input structures a, b to the visit function.
         visit = functools.partial(_EqCompareNested._visit, a=a, b=b)
 
         try:
-            iterutils.remap(a, visit=visit, reraise_visit=True)
+            iterutils.remap(
+                a, visit=visit, reraise_visit=True, enter=_EqCompareNested._enter
+            )
```

Comment:
the `_enter` function should probably have been used here before, which confused me a little

Reply:
I guess you're right. I haven't looked into the code yet.

Reply:
the behavior should be more consistent overall now
`set` comparison behavior is now listed under `compare_funcs` which prevents entering/traversing the items